gcs/deleteFile/deleteFile.go

```diff
@@ -0,0 +1,160 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Evan Spendlove (@evanSpendlove)
+//
+// DeleteFile is a probe that deletes a file from GCS.
+// It tests the file deletion aspect of GCS and ensures that the file
+// is deleted after the deletion request is made.
+
+package gcs.deletefile
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+
+	"cloud.google.com/go/storage"
+)
+
+// Probe holds aggregate information about all probe runs, per-target.
+type Probe struct {
+	name    string
+	c       *ProbeConf
+	targets []string
+	opts    *options.Options
+
+	res map[string]*metrics.EventMetrics // Results by target
+	l   *logger.Logger
+}
+
+// Init initializes the probe with the given params.
+func (p *Probe) Init(name string, opts *options.Options) error {
+	c, ok := opts.ProbeConf.(*ProbeConf)
+	if !ok {
+		return fmt.Errorf("not a gcs.deleteFile config")
+	}
+	p.c = c // Assign config
+	p.name = name
+	p.opts = opts
+	p.l = opts.Logger
+	p.res = make(map[string]*metrics.EventMetrics)
+
+	return nil
+}
+
+// Start starts and runs the probe indefinitely.
+func (p *Probe) Start(ctx context.Context, dataChan chan *metrics.EventMetrics) {
+	probeTicker := time.NewTicker(p.opts.Interval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			probeTicker.Stop()
+			return
+		case <-probeTicker.C:
+			// On probe tick, write data to the channel and run probe.
+			for _, em := range p.res {
+				dataChan <- em
+			}
+
+			// TODO: CHANGE THIS BIT TO OUR PROBE IMPLEMENTATION
+			p.targets = endpoint.NamesFromEndpoints(p.opts.Targets.ListEndpoints())
+			p.initProbeMetrics()
+			probeCtx, cancelFunc := context.WithDeadline(ctx, time.Now().Add(p.opts.Timeout))
+			p.runProbe(probeCtx)
+			cancelFunc()
+		}
+	}
+}
+
+// TODO: Change this!!
+// initProbeMetrics initializes missing probe metrics.
+func (p *Probe) initProbeMetrics() {
+	for _, target := range p.targets {
+		if p.res[target] != nil {
+			continue
+		}
+		var latVal metrics.Value
+		if p.opts.LatencyDist != nil {
+			latVal = p.opts.LatencyDist.Clone()
+		} else {
+			latVal = metrics.NewFloat(0)
+		}
+		p.res[target] = metrics.NewEventMetrics(time.Now()).
+			AddMetric("total", metrics.NewInt(0)).
+			AddMetric("success", metrics.NewInt(0)).
+			AddMetric("latency", latVal).
+			AddLabel("ptype", "redis").
+			AddLabel("probe", p.name).
+			AddLabel("dst", target)
+	}
+}
+
+// TODO: Improve this by passing the storage client variable/making it accessible to reduce num of conns.
+// TODO: CHANGE THIS - THIS IS THE MAIN PROBE RUN() function
```

Comment:
TODO: CHANGE THIS is not massively clear about what your'e changing :-)

Reply:
Agreed, this file was not meant to be in this PR.

I have removed the deleteFile.go from this PR, as I want to merge it in a different PR.